File: problems/451_sort_characters_by_frequency/Solution.java
import java.util.*;

class Solution {
    public String frequencySort(String s) {
        int[] count = new int[128];
        for (char c : s.toCharArray()) count[c]++;
        List<Character> chars = new ArrayList<>();
        for (int i = 0; i < 128; i++) if (count[i] > 0) chars.add((char) i);
        chars.sort((a, b) -> count[b] - count[a]);
        StringBuilder sb = new StringBuilder();
        for (char c : chars) for (int i = 0; i < count[c]; i++) sb.append(c);
        return sb.toString();
    }
}

File: problems/451_sort_characters_by_frequency/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> count;
        for (char c : s) count[c]++;
        sort(s.begin(), s.end(), [&](char a, char b) {
            return count[a] == count[b] ? a < b : count[a] > count[b];
        });
        return s;
    }
};
